src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,147 @@
+package nextstep.subway.line.domain;
+
+import nextstep.subway.station.domain.Station;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.*;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new LinkedList<>();
+
+    public Sections() {
+    }
+
+    public Sections(List<Section> sections) {
+        sections = sections;
+    }
+
+    /* 도메인 기능 메서드 */
+
+    // 노선의 역 조회
+    public List<Station> getStations() {
+        List<Station> stations = new ArrayList<>();
+        if(getSectionSize() > 0){
+            Station downStation = getFirstStation();
+            stations.add(downStation);
+
+            while (downStation != null) {
+                Station finalDownStation = downStation;
+                Optional<Section> nextLineStation = sections.stream()
+                        .filter(it -> it.getUpStation() == finalDownStation)
+                        .findFirst();
+                if (!nextLineStation.isPresent()) {
+                    break;
+                }
+                downStation = nextLineStation.get().getDownStation();
+                stations.add(downStation);
+            }
+        }
+        return stations;
+    }
+
+    // 구간개수
+    public int getSectionSize() {
+        return sections.size();
+    }
+
+    // 상행역 종점 조회
+    public Station getFirstStation() {
+        return Optional.ofNullable(sections.get(0))
+                .map(it -> it.getUpStation())
+                .orElse(null);
+    }
+
+    // 하행역 종점 조회
+    public Station getLastStation() {
+        return Optional.ofNullable(sections.get(getSectionSize()-1))
+                .map(it -> it.getDownStation())
+                .orElse(null);
+    }
+
+    // 구간 추가
+    public void add(Section section) {
+        if (getStations().size() == 0) {
+            sections.add(section);
+            return;
+        }
+
+        // 벨리데이션
+        isConnectableSection(section);
+        isAlreadyAdded(section);
```

Comment:
```suggestion
        checkConnectableSection(section);
        validateAlreadyAdded(section);
```

is나 has 같은 메소드 네이밍은 boolean으로 여부를 리턴해줄 것 같은 네이밍이라, check 혹은 validate 같은 단어로 바꿔 표현해보면 어떨까요? :)